src/gallery/gestures.js

```diff
@@ -126,8 +126,9 @@ const zoomScroll = ( e, renderNext, items, current ) => {
 	const image = grabImageFromEvent( e )
 	const transform = image.style.transform
 	const scale = transform ? grabScaleFromTransform( transform ) : scaleMin
+	const isImgLandscape = image.naturalHeight <= image.naturalWidth
 	const horizontalLimit = clientWidth / 2
-	const verticalLimit = clientHeight / 2
+	const verticalLimit = isImgLandscape ? clientHeight / 8 : clientHeight / 2
```

Comment:
can you explain this line? especially the `8` magic number 🤔

Reply:
I ended up leaving the image `transform-origin` as the default `center`. What that means is that once the image is enlarged, all the `translate3d` coordinates start from the center as zero (`(0, 0, 0)`). If we visualize that for the `horizontalLimit` for example, then from the center to right/left the distance "available" to scroll within the viewport is the `clientWidth` divided by two (one half to the right, one to left). 

Now let's take that same logic to the `verticalLimit`. If the image is a portrait, then `clientHeight / 2` will give us all of the available distance from the top/bottom halfs to scroll freely and see all of the zoomed in image. However if the image has a landscape orientation then using all of the distance available results in the unwanted behavior that Stephane pointed out: the bottom or top of the image could reach the middle of the screen leaving an awkward, empty space visible. The idea I'm proposing here there is to set the vertical limit to a lower value if the image is landscape so that we can restrain how much distance is available to scroll. Why 8? It's just the value that worked better after trying out different numbers (so we could find tune that number if needed). Does that make sense?